### PR TITLE
Generate build logs to show PIE

### PR DESCRIPTION
**Pull request by RandySheriffH**
Set CMAKE_EXPORT_COMPILE_COMMANDS to get build log to prove PIE has been used for Mac binary generation.

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +30,8 @@ set(CMAKE_CXX_STANDARD 14)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 # NOTE: POSITION INDEPENDENT CODE hurts performance, and it only make sense on POSIX systems
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+# Generate build logs listing all flags used
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```
Comment (snnn) [inline cmake/CMakeLists.txt:34]: This  will affect all the builds, no matter it is windows, linux or mac.

Comment (RandySheriffH) [reply cmake/CMakeLists.txt:34]: Yes it is, but all it does is to generate a json file of build commands ever issued, no impact on output binaries or build time.

Comment (edgchen1) [reply cmake/CMakeLists.txt:34]: FYI, I set this from build.sh:
`build.sh ... --cmake_extra_defines CMAKE_EXPORT_COMPILE_COMMANDS=1`

That's another way to do it if we don't want it on by default.

Comment (pranavsharma) [reply cmake/CMakeLists.txt]: Can we do this for Mac only since the PIE requirement was for Mac only?

Comment (RandySheriffH) [reply cmake/CMakeLists.txt]: Done - moved to "if Darwin...".

File: cmake/CMakeLists.txt [hunk 2]
```diff
@@ -293,6 +293,7 @@ if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     #For Mac compliance
     message("Adding flags for Mac builds")
     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector-strong")
+    set (CMAKE_VERBOSE_MAKEFILE ON)
```
Comment (pranavsharma) [inline cmake/CMakeLists.txt:296]: Should this be CMAKE_EXPORT_COMPILE_COMMANDS?

**Pull request closed without merging**